ToDoTime/ClientApp/src/pages/newList/NewList.tsx

```diff
@@ -1,107 +1,84 @@
 // import Select from "react-select";
 import ReactSelect from "react-select-virtualized";
 import { BoxCustomIcon } from "shared/BoxCustomIcon";
-import { AvailableIconsEnum } from "shared/availableIcons";
 import { Input } from "shared/components/Input";
 import { MainContainerTemplate } from "shared/components/MainContainerTemplate";
 import { APP_GOLD_COLOR } from "shared/constants/appStyles";
-import styles from "./NewList.module.scss";
-
-interface IIconOption {
-  value: string;
-  label: string;
-}
+import { useNewListForm } from "./useNewListForm";
+import { TextArea } from "shared/components/TextArea";
+import { Button } from "shared/components/Button";
 
 export const NewList = () => {
-  const [selectedIcon, setSelectedIcon] = useState<IIconOption>();
-  const options = useMemo(() => {
-    const innerOptions: IIconOption[] = [];
-    Object.keys(AvailableIconsEnum).forEach((k) => {
-      // Remove the first two characters
-      const trimmedString = k.slice(2);
-
-      // Split the string whenever an uppercase letter is found
-      const formattedString = trimmedString.replace(/([A-Z])/g, " $1");
-
-      innerOptions.push({
-        value: k,
-        label: formattedString.trim()
-      });
-    });
-
-    return innerOptions;
-  }, []);
+  const { nameInputProps, iconSelectProps, descriptionAreaProps } = useNewListForm();
 
   return (
     <MainContainerTemplate title="New List">
       <FormGrid>
         <Field>Name:</Field>
         <Value>
-          <Input />
+          <Input {...nameInputProps} />
         </Value>
 
         <Field>Icon</Field>
         <Value>
           {/* Useful to check for the props on react-select-virtualized since the lib doesnt have types */}
           {/* <Select /> */}
-          <ReactSelect
-            value={selectedIcon}
-            onChange={(option: IIconOption) => {
-              setSelectedIcon(option);
-            }}
-            options={options}
-            formatOptionLabel={(option: IIconOption) => {
-              return (
-                <IconOptionRendered>
-                  <BoxCustomIcon
-                    nameIcon={option.value}
-                    propsIcon={{ size: 25, style: { marginRight: 10 } }}
-                  />
-                  {option.label}
-                </IconOptionRendered>
-              );
-            }}
-            classNames={{
-              container: () => styles["new-list-icon-select-container"],
-              menu: () => styles["new-list-icon-select-menu"]
-            }}
-          />
-          {selectedIcon && (
+          <ReactSelect {...iconSelectProps} />
+          {iconSelectProps.value && (
             <BoxCustomIcon
-              nameIcon={selectedIcon.value}
+              nameIcon={iconSelectProps.value.value}
               propsIcon={{ size: 35, color: APP_GOLD_COLOR, style: { marginLeft: 10 } }}
             />
           )}
         </Value>
+        <Field>Description</Field>
```

Comment:
We should leave description for the tasks, not much use having it for the list, or at least make it optional.

Reply:
Made it optional.